receiver/flinkmetricsreceiver/process.go

```diff
@@ -0,0 +1,204 @@
+				_ = s.mb.RecordFlinkmetricsTaskmanagerStatusJvmMemoryHeapMaxDataPoint(now, metric.Value)
+			case "Status.JVM.Memory.Heap.Used":
+				_ = s.mb.RecordFlinkmetricsTaskmanagerStatusJvmMemoryHeapUsedDataPoint(now, metric.Value)
+			}
+		}
+	}
+	if len(taskmanagerMetricInstances) > 0 {
+		s.mb.EmitForResource(metadata.WithHost(taskmanagerMetricInstances[0].Host), metadata.WithTaskmanagerID(taskmanagerMetricInstances[0].TaskmanagerID))
+	}
+}
+
+func (s *flinkmetricsScraper) processJobsMetrics(now pcommon.Timestamp, jobsMetricsInstances []*models.JobMetrics) {
+	for _, jobsMetrics := range jobsMetricsInstances {
+		for _, metric := range jobsMetrics.Metrics {
+			switch metric.ID {
+			case "numRestarts":
+				_ = s.mb.RecordFlinkmetricsJobRestartCountDataPoint(now, metric.Value)
+			case "lastCheckpointSize":
+				_ = s.mb.RecordFlinkmetricsJobLastCheckpointSizeDataPoint(now, metric.Value)
+			case "lastCheckpointDuration":
+				_ = s.mb.RecordFlinkmetricsJobLastCheckpointTimeDataPoint(now, metric.Value)
+			case "numberOfInProgressCheckpoints":
+				_ = s.mb.RecordFlinkmetricsJobCheckpointsCountDataPoint(now, metric.Value, metadata.AttributeCheckpointInProgress)
+			case "numberOfCompletedCheckpoints":
+				_ = s.mb.RecordFlinkmetricsJobCheckpointsCountDataPoint(now, metric.Value, metadata.AttributeCheckpointCompleted)
+			case "numberOfFailedCheckpoints":
+				_ = s.mb.RecordFlinkmetricsJobCheckpointsCountDataPoint(now, metric.Value, metadata.AttributeCheckpointFailed)
+			}
+		}
+	}
+	if len(jobsMetricsInstances) > 0 {
+		s.mb.EmitForResource(metadata.WithHost(jobsMetricsInstances[0].Host), metadata.WithJobName(jobsMetricsInstances[0].JobName))
+	}
+}
+
+func (s *flinkmetricsScraper) processSubtaskMetrics(now pcommon.Timestamp, subtaskMetricsInstances []*models.SubtaskMetrics) {
+	for _, subtaskMetrics := range subtaskMetricsInstances {
+		for i, metric := range subtaskMetrics.Metrics {
+			switch {
+			// record task metrics
+			case metric.ID == "numRecordsIn":
+				_ = s.mb.RecordFlinkmetricsTaskRecordCountDataPoint(now, metric.Value, metadata.AttributeRecordIn)
+			case metric.ID == "numRecordsOut":
+				_ = s.mb.RecordFlinkmetricsTaskRecordCountDataPoint(now, metric.Value, metadata.AttributeRecordOut)
+			case metric.ID == "numLateRecordsDropped":
+				_ = s.mb.RecordFlinkmetricsTaskRecordCountDataPoint(now, metric.Value, metadata.AttributeRecordLateRecordsDropped)
+			}
+			if i == len(subtaskMetrics.Metrics)-1 {
+				s.mb.EmitForResource(metadata.WithHost(subtaskMetrics.Host), metadata.WithTaskmanagerID(subtaskMetrics.TaskmanagerID), metadata.WithJobName(subtaskMetrics.JobName), metadata.WithTaskName(subtaskMetrics.TaskName), metadata.WithSubtaskIndex(subtaskMetrics.SubtaskIndex))
+			}
+		}
+	}
+
+	for _, subtaskMetrics := range subtaskMetricsInstances {
+		for i, metric := range subtaskMetrics.Metrics {
+			switch {
+			// record operator metrics
+			case strings.Contains(metric.ID, ".numRecordsIn"):
```

Comment:
I can't find the format that requires splitting here. Can you give an example?

Reply:
sure, in the apiresponses folder there is a `subtask_metrics_values.json` . In the file a task and operator metrics exist that are collected using the subtasks endpoint. You can tell which are operator metrics because they have a <operator_name>.metric, like `Source__Custom_Source.numRecordsOut` whereas the task would just be `numRecordsOut`.  Since the `Source__Custom_Source` can have any operator name the best way to distinguish an operators is by looking for the dot metric name.